### PR TITLE
sapmexporter: Add option to log ingest response

### PR DESCRIPTION
**Pull request by asuresh4**
**Description:** Add option to log detailed ingest response in `sapmexporter`.

**Testing:** Added unit test

**Documentation:** Added to README

### CODE REVIEW

File: exporter/sapmexporter/exporter.go [hunk 1]
```diff
@@ -104,6 +105,7 @@ func newSAPMTracesExporter(cfg *Config, set component.ExporterCreateSettings) (c
 // and the last experienced error if any translation or export failed
 func (se *sapmExporter) pushTraceData(ctx context.Context, td ptrace.Traces) error {
 	rss := td.ResourceSpans()
+	td.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0).SetSpanID(pcommon.InvalidSpanID())
```
Comment (dmitryax) [inline exporter/sapmexporter/exporter.go:108]: Is this a debug line?

Comment (asuresh4) [reply exporter/sapmexporter/exporter.go]: That was something I added while testing. Thanks for catching it. Removed. 

Also, updated `sapm-proto` to v0.11.0.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased
+- `sapmexporter`: Add config option to log responses from Splunk APM. (#11425)
```
Comment (dmitryax) [inline CHANGELOG.md:4]:
```suggestion

- `sapmexporter`: Add config option to log responses from Splunk APM. (#11425)
```

File: exporter/sapmexporter/exporter.go [hunk 3]
```diff
@@ -18,7 +18,6 @@ package sapmexporter // import "github.com/open-telemetry/opentelemetry-collecto
 import (
 	"context"
 	"errors"
-
```
Comment (dmitryax) [inline exporter/sapmexporter/exporter.go:21]: Looks like this causes lint failures

File: exporter/sapmexporter/README.md [hunk 4]
```diff
@@ -50,6 +52,7 @@ exporters:
     endpoint: https://ingest.YOUR_SIGNALFX_REALM.signalfx.com/v2/trace
     max_connections: 100
     num_workers: 8
+    log_detailed_response_from_trace_ingest: true
```
Comment (dmitryax) [inline exporter/sapmexporter/README.md:55]: Even if we want this option, let's make it shorter `log_detailed_response`? I seems too verbose

Comment (asuresh4) [reply exporter/sapmexporter/README.md]: Updated

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@asuresh4 looks like TestClientConcurrentCompaction is always failing. Can you rebase to pick up https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/11054

**Review by codeboten — COMMENTED** [review]
Please rebase once again

**Comment by asuresh4** [comment]
@codeboten @dmitryax rebased with latest from main.

**Comment by dmitryax** [comment]
@asuresh4 Sorry it needs another rebase :(

**Comment by asuresh4** [comment]
Ok, fixed conflicts again. Hopefully we can get it merged this time 😬

**Review by codeboten — APPROVED** [review]

**Pull request merged**